addons/stock/models/stock_quant.py

```diff
@@ -77,7 +77,7 @@ def check_product_id(self):
     def check_quantity(self):
         for quant in self:
             if float_compare(quant.quantity, 1, precision_rounding=quant.product_uom_id.rounding) > 0 and quant.lot_id and quant.product_id.tracking == 'serial':
-                raise ValidationError(_('A serial number should only be linked to a single product.'))
+                raise ValidationError(_('A serial number should only be linked to a single product. Serial %s of product %s would have a quantity of %f.') % (quant.lot_id.name, quant.product_id.name, quant.quantity)
```

Comment:
in stable you cannot change the translation.
solution:
```python
msg = _('A serial number should only be linked to a single product.')
msg += _('Serial %s of product %s would have a quantity of %f.') % (quant.lot_id.name, quant.product_id.display_name, quant.quantity)
raise ValidationError(msg)
```

note it is better to user `display_name` for `product_id`

Reply:
thanks, fixed.